### PR TITLE
Update Decent CI

### PR DESCRIPTION
**Pull request by Myoldmopar**
Decent CI VMs were failing like crazy.  I spent time getting new images up and running.  It's on Ubuntu 24.04 now, so we are good for several years on the config side.  This brings in the random pieces needed to get it happy...

FWIW, there were also a few tweaks on the Decent CI side to get things working nicely: https://github.com/Myoldmopar/decent_ci/compare/011c52bc251fee7fd1f0fc5480edb712d5c45e89..c3c6ea6b4c993bc98be4d109d237e6111f5db7e8

### CODE REVIEW

File: .decent_ci-Linux.yaml [hunk 1]
```diff
@@ -1,13 +1,14 @@
 compilers:
   - name: "gcc"
-    version: "11.4"
+    version: "13.2"
```
Comment (Myoldmopar) [inline .decent_ci-Linux.yaml:3]: Compiler update, default on Ubuntu 24.04 is 13.2.

File: .decent_ci-Linux.yaml [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
     cmake_extra_flags: -DLINK_WITH_PYTHON:BOOL=ON -DPYTHON_CLI:BOOL=OFF -DPython_REQUIRED_VERSION:STRING=3.12.2 -DPython_ROOT_DIR:PATH=~/.pyenv/versions/3.12.2/ -DBUILD_FORTRAN:BOOL=ON -DBUILD_TESTING:BOOL=ON -DENABLE_GTEST_DEBUG_MODE:BOOL=OFF -DBUILD_PERFORMANCE_TESTS:BOOL=ON -DVALGRIND_ANALYZE_PERFORMANCE_TESTS:BOOL=ON -DENABLE_PCH:BOOL=OFF
     collect_performance_results: true
     skip_regression: true
     s3_upload_bucket: energyplus
+    num_parallel_builds: 18
```
Comment (Myoldmopar) [inline .decent_ci-Linux.yaml:8]: The CI machine has 20 cores, but only 16 GB of RAM.  It was swapping, or getting close to swapping, so I explicitly pulled the number down here just a little.

File: .decent_ci.yaml [hunk 3]
```diff
@@ -2,7 +2,7 @@ results_repository : Myoldmopar/EnergyPlusBuildResults
 results_path : _posts
 results_base_url : https://myoldmopar.github.io/EnergyPlusBuildResults
 regression_repository : NREL/EnergyPlusRegressionTool
-regression_branch : BumpToBoto3 # this is the branch of NREL/EnergyPlusRegressionTool to use (usually main)
+regression_branch : main # this is the branch of NREL/EnergyPlusRegressionTool to use (usually main)
```
Comment (Myoldmopar) [inline .decent_ci.yaml:5]: This should have been `main` a while back, as the `BumpToBoto3` branch was merged into main already.

File: third_party/fmt-8.0.1/include/fmt/format.h [hunk 4]
```diff
@@ -316,8 +316,10 @@ template <> constexpr auto num_bits<fallback_uintptr>() -> int {
 
 FMT_INLINE void assume(bool condition) {
   (void)condition;
-#if FMT_HAS_BUILTIN(__builtin_assume)
```
Comment (Myoldmopar) [inline third_party/fmt-8.0.1/include/fmt/format.h:319]: There was a `fmt` compiler warning being emitted.  I had seen this locally for a while, but I could just look past it.  Unfortunately the Decent CI script saw this as a failure, so it was time to fix it.  There are some notes in here for the change, but basically, it seems it's a GCC false positive issue.  This brings in a cheap workaround to the warning.  I'm open to other ways to do this, but I'm not waiting to change it for this PR.  Feel free to open a new PR that reverts this and does it differently.

**Review by Myoldmopar — COMMENTED** [review]
Alright, I've already shut down the wild Ubuntu 22 machines, so this needs to go into develop right away.  Results look perfect now after all the tweaks, so once GHA gets done with it, I'll get it merged in.

**Comment by Myoldmopar** [comment]
Oh, and you can even see that fmt warning on the Ubuntu 24 GHA machines of course.  https://github.com/NREL/EnergyPlus/actions/runs/11895410202/job/33144923634#step:9:294.

Probably need to add a g++ output parser to the GHA builds to catch warnings gracefully.  (@jmarrec any interest there?)

**Comment by Myoldmopar** [comment]
Good to go.  I'll pull develop into a few branches to get fresh results and get things moving.

**Comment by jmarrec** [comment]
> Oh, and you can even see that fmt warning on the Ubuntu 24 GHA machines of course. https://github.com/NREL/EnergyPlus/actions/runs/11895410202/job/33144923634#step:9:294.
> 
> Probably need to add a g++ output parser to the GHA builds to catch warnings gracefully. (@jmarrec any interest there?)

* Related: #10596
* Sure, we can add a problem matcher

**Comment by jmarrec** [comment]
* #10832

**Pull request merged**